packages/geo/src/Providers/AmazonLocationServicesProvider.ts

```diff
@@ -102,4 +116,95 @@ export class AmazonLocationServicesProvider implements GeoProvider {
 
 		return { mapName, style };
 	}
+
+	/**
+	 * Search by text input with optional parameters
+	 * @param  {string} text - The text string that is to be searched for
+	 * @param  {SearchByTextOptions} options? - Optional parameters to the search
+	 * @returns {Promise<Place[]>} - Promise resolves to a list of Places that match search parameters
+	 */
+	public async searchByText(
+		text: string,
+		options?: SearchByTextOptions
+	): Promise<Place[]> {
+		try {
+			const credentialsOK = await this._ensureCredentials();
+			if (!credentialsOK) {
+				return Promise.reject('No credentials');
+			}
+		} catch (error) {
+			throw '';
+		}
+
+		/**
+		 * Setup the searchInput
+		 */
+		const locationServicesInput: SearchPlaceIndexForTextCommandInput = {
+			Text: text,
+			IndexName: this._config.place_indexes.default,
+		};
+
+		/**
+		 * Map search options to Location Services input object
+		 */
+		if (options) {
+			locationServicesInput.FilterCountries = options.countries;
+			locationServicesInput.MaxResults = options.maxResults;
+
+			if (options.placeIndexName) {
+				locationServicesInput.IndexName = options.placeIndexName;
+			}
+
+			if (options['biasPosition']) {
+				locationServicesInput.BiasPosition = options['biasPosition'];
+			} else if (options['searchAreaConstraints']) {
+				locationServicesInput.FilterBBox = options['searchAreaConstraints'];
+			}
```

Comment:
can we use the same way to access properties as above? e.g
```suggestion
			if (options.biasPosition) {
				locationServicesInput.BiasPosition = options.biasPosition;
			} else if (options.searchAreaConstraints) {
				locationServicesInput.FilterBBox = options.searchAreaConstraints;
			}
```

Reply:
This was what I had to use due using a union type here for the `options`.

If I use the `object.key` syntax here, it throws issues saying:
```
Property 'biasPosition' does not exist on type 'SearchByTextOptions'.
  Property 'biasPosition' does not exist on type 'SearchByTextOptionsWithSearchAreaConstraints'.
```
```
Property 'searchAreaConstraints' does not exist on type 'SearchByTextOptions'.
  Property 'searchAreaConstraints' does not exist on type 'SearchByTextOptionsWithBiasPosition'.
```

However, using the `object["key"]` syntax, it does not throw.

Reply:
Can we extend `SearchByTextOptions` to accept those properties?

Reply:
@wlee221 and I spoke offline about this and we're going to leave it like this for now and possibly update how this is done later.